lib/ui/text.dart

```diff
@@ -431,6 +431,85 @@ enum TextDecorationStyle {
   wavy
 }
 
+/// Defines how the paragraph will handle the ascent of the first line and
+/// descent of the last line. These lines are referred to as "boundary" lines.
+class BoundaryLineHeightBehavior {
+
+  /// Creates a new BoundaryLineHeightBehavior object.
+  ///
+  ///  * first: When true, the [TextStyle.height] modifier will be applied to
+  ///    to the ascent of the first line. When false, the font's default ascent
+  ///    will be used.
+  ///  * last: When true, the [TextStyle.height] modifier will be applied to
+  ///    to the descent of the last line. When false, the font's default descent
+  ///    will be used.
+  ///
+  /// All properties default to true (height modificaitons applied as normal).
+  BoundaryLineHeightBehavior({
+    this.first = true,
+    this.last = true,
+  });
+
+  /// Creates a new BoundaryLineHeightBehavior object from an encoded form.
+  ///
+  /// See [encode] for the creation of the encoded form.
+  BoundaryLineHeightBehavior.fromEncoded(int encoded) : first = (encoded & 0x1) > 0,
+                                                        last = (encoded & 0x2) > 0;
+
+
+  /// Whether to apply the [TextStyle.height] modifier or not to the ascent of
+  /// the first line in the paragraph.
+  ///
+  /// When true, the [TextStyle.height] modifier will be applied to to the ascent
+  /// of the first line. When false, the font's default ascent will be used and
+  /// the [TextStyle.height] will have no effect on the ascent of the first line.
+  ///
+  /// Defaults to true (height modificaitons applied as normal).
+  final bool first;
+
+  /// Whether to apply the [TextStyle.height] modifier or not to the descent of
+  /// the last line in the paragraph.
+  ///
+  /// When true, the [TextStyle.height] modifier will be applied to to the descent
+  /// of the last line. When false, the font's default descent will be used and
+  /// the [TextStyle.height] will have no effect on the descent of the last line.
+  ///
+  /// Defaults to true (height modificaitons applied as normal).
+  final bool last;
+
+  /// Returns an encoded int representation of this object.
+  int encode() {
+    return 0 + (first ? 1 << 0 : 0) + (last ? 1 << 1 : 0);
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is BoundaryLineHeightBehavior
+        && other.first == first
+        && other.last == last;
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      first,
+      last,
+    );
+  }
+
+  @override
+  String toString() {
+    return 'BoundaryLineHeightBehavior('
+             'first: $first, '
+             'last: $last, '
```

Comment:
when you write a test for the toString, you'll see you have a trailing comma. :-)